src/Symfony/Bundle/SecurityBundle/DependencyInjection/SecurityExtension.php

```diff
@@ -518,7 +518,17 @@ private function createEncoder($config, ContainerBuilder $container)
         // Argon2i encoder
         if ('argon2i' === $config['algorithm']) {
             if (!Argon2iPasswordEncoder::isSupported()) {
-                throw new InvalidConfigurationException('Argon2i algorithm is not supported. Please install the libsodium extension or upgrade to PHP 7.2+.');
+                if (\PHP_VERSION_ID >= 70200) {
+                    if (\defined('SODIUM_LIBRARY_VERSION')) {
+                        $sodiumVersion = explode('.', \SODIUM_LIBRARY_VERSION);
+                        $sodiumVersion = ($sodiumVersion[0] * 10000 + $sodiumVersion[1] * 100 + $sodiumVersion[2]);
```

Comment:
What purpose do these parenthesis serve?

Reply:
Keeping it all together, if they're unneccesary i'll remove them.